### DISCUSSION TITLE
How to specify where a world is saved?

### DISCUSSION THREAD

**Discussion by ryansheehan** [Pryaxis/TShock#1835] [Q&A]
After this [commit](https://github.com/Pryaxis/TShock/commit/7ad46abced17113ca4338598c73c172828e43355) I can no longer specify where worlds get saved.  Is there a new way to specify where to save a world when it gets created?

**Comment by 4dnr3w** [comment]
bumped as I am struggling to put my previous .wld in to the updated docker

**Comment by ryansheehan** [comment]
As far as I can tell, the world path is now hard coded.  It was nice to be able to group the save file with the config.

**Reply by hakusaro** [reply]
It was removed because it was hard crashing instead of working.

**Comment by ColinBohn** [comment]
It looks like the commit you linked has some advice:
`Removed -worldpath because you should now just use -world with the full file path`

Is using -world with the full path to the world file not currently working?

**Reply by ryansheehan** [reply]
The problem with `-world` is that if a world does not exist the server still tries to start and just hangs.  The nice thing about `-worldpath`, especially for a docker setup, was that it provided a direction for where new worlds get created and can be volumed into the host machine.  It's a bit more of an inconvenience now for docker because the command becomes quite large having to directly specify the fully qualified path.

It's not the end of the world.  I have published a new container and instructions to specify the full hardcoded path.  It's just a little more painful to use.  Not sure why hard coding the path to .wld files  does to improve the server.

**Reply by hakusaro** [reply]
@ryansheehan it was implemented at the TShock level but it should really have been lower in the API. More broadly we’d love to have an official Docker process. If you want to contribute your Dockerfile and port a build system to actions we would be more than happy to carry on with maintenance.

**Reply by QuiCM** [reply]
> 
> 
> The problem with `-world` is that if a world does not exist the server still tries to start and just hangs. The nice thing about `-worldpath`, especially for a docker setup, was that it provided a direction for where new worlds get created and can be volumed into the host machine. It's a bit more of an inconvenience now for docker because the command becomes quite large having to directly specify the fully qualified path.
> 
> It's not the end of the world. I have published a new container and instructions to specify the full hardcoded path. It's just a little more painful to use. Not sure why hard coding the path to .wld files does to improve the server.

I believe if you use -world as well as the -autocreate(?) flag then it will generate a new world should the path specified not exist. If the world does exist, it will load it. Not exactly intuitive unfortunately, but the best we can get for the moment

**Comment by ryansheehan** [comment]
Unfortunately the autocreate flag misses a lot of the options the
interactive provides.  For now, I've updated the container and docs for
proper volume setup.  I'll take a look at creating a container directly
from source to contribute.
